Client/src/Components/DatasetUpload/MetaSection/MetaForm.tsx

```diff
@@ -1,19 +1,30 @@
 import { Box, Grid, Typography } from '@material-ui/core'
-import { FastField, FieldArray } from 'formik'
+import { FastField, Field, FieldArray } from 'formik'
+import { MuiSelectFormik, MuiTextFieldFormik } from '../../Forms/FormikFields'
 
 import { IMaterial } from '../../../Models/Datasets/IDatasetModel'
 import { MaterialSelectChipArray } from './MaterialSelectChipArray'
-import { MuiTextFieldFormik } from '../../Forms/FormikFields'
 import React from 'react'
 import { classStyles } from '../../../appTheme'
 import { get } from 'lodash'
 
 interface IProps {
   materials: IMaterial[],
+  categories: any[],
+  subcategories: any[],
+}
+
+const getOptions = (options: any[]): any => {
+  return (
+    <>
+      <option aria-label="None" value="" />
+      {options.map(option => <option key={option.id} value={option.id}> {option.name} </option>)}
+    </>
+  )
 }
 
 export const MetaForm = (props: IProps) => {
-  const { materials } = props
+  const { materials, categories, subcategories } = props
```

Comment:
Do you think it would be better to use Context instead of prop drilling?

Reply:
honestly that is really not a bad idea, cause once they are loaded they dont have to be refetched on other pages, but at the same time, what if they change, i think a proper solution would be to use a store lib, Ill keep in mind for the future, and about the prop drilling, i sorta wanted the MetaForm component to be dumb and not have any logic, so put the calls here, Ill leave it like this for now, but will return to this later next time a situation like this occurs